src/state_machine.zig

```diff
@@ -592,9 +687,131 @@ pub fn StateMachineType(
+
+            if (self.get_scan_from_filter(filter)) |scan| {
+                var scan_buffer = std.mem.bytesAsSlice(Transfer, self.scan_buffer);
+                assert(scan_buffer.len >= constants.batch_max.get_account_transfers);
+
+                var scan_lookup = self.scan_lookup.get(.transfer);
+                scan_lookup.* = TransfersScanLookup.init(
+                    &self.forest.grooves.transfers,
+                    scan,
+                );
+
+                // Limiting the buffer size according to the query limit.
+                const limit = @min(filter.limit, constants.batch_max.get_account_transfers);
+                scan_lookup.read(
+                    scan_buffer[0..limit],
+                    &prefetch_get_account_transfers_callback,
+                );
+            } else {
+                // TODO(batiati): Improve the way we do validations on the state machine.
+                log.info("invalid filter for get_account_transfers: {any}", .{filter});
+                self.forest.grid.on_next_tick(
+                    &prefetch_scan_next_tick_callback,
+                    &self.scan_next_tick,
+                );
+            }
+        }
+
+        fn prefetch_get_account_transfers_callback(scan_lookup: *TransfersScanLookup) void {
+            const self: *StateMachine = ScanLookup.parent(.transfer, scan_lookup);
+            self.scan_result_count = @intCast(scan_lookup.slice().len);
+
+            self.scan_lookup = .null;
+            self.forest.scan_buffer_pool.reset();
+            self.forest.grooves.transfers.scan_builder.reset();
+
+            self.prefetch_finish();
+        }
+
+        fn prefetch_get_account_history(self: *StateMachine, filter: AccountFilter) void {
+            self.forest.grooves.accounts.prefetch_enqueue(filter.account_id);
+            self.forest.grooves.accounts.prefetch(
+                prefetch_get_account_history_lookup_account_callback,
+                self.prefetch_context.get(.accounts),
+            );
+        }
+
+        fn prefetch_get_account_history_lookup_account_callback(
+            completion: *AccountsGroove.PrefetchContext,
+        ) void {
+            const self: *StateMachine = PrefetchContext.parent(.accounts, completion);
+            self.prefetch_context = .null;
+
+            const filter = parse_filter_from_input(self.prefetch_input.?);
+            self.prefetch_get_account_history_scan(filter);
+        }
+
+        fn prefetch_get_account_history_scan(self: *StateMachine, filter: AccountFilter) void {
             assert(self.scan_result_count == 0);
 
+            if (self.get_scan_from_filter(filter)) |scan| {
+                if (self.forest.grooves.accounts.get(filter.account_id)) |account| {
+                    if (account.flags.history) {
+                        var scan_buffer = std.mem.bytesAsSlice(
+                            AccountHistoryGrooveValue,
+                            self.scan_buffer,
+                        );
```

Comment:
:thinking: depending on the multiple of `get_account_transfers`, is it possible that `scan_buffer`'s length is not an even multiple of `@sizeOf(AccountHistoryGrooveValue)`?